test/services/katello/registration_manager_test.rb

```diff
@@ -71,6 +71,24 @@ def test_existing_host_mismatch_uuid
           assert @klass.validate_hosts(hosts, @org, @host.name, 'different-uuid')
         end
 
+        def test_existing_host_already_registered
```

Comment:
If you go with my approach above, this test would become something like:

```
        def test_re_register_build_mode                                                                                                                                                       
          @host.build = false                                                                                                                                                                 
          Setting[:host_re_register_build_only] = true                                                                                                                                        
                                                                                                                                                                                              
          error = assert_raises(Katello::Errors::RegistrationError) { @klass.validate_hosts(hosts, @org, @host.name) }                                                                        
          assert_match(/existing registration/, error.message)                                                                                                                                
                                                                                                                                                                                              
          @host.build = true                                                                                                                                                                  
          assert @klass.validate_hosts(hosts, @org, @host.name, 'existing_system_uuid')                                                                                                       
        end 
```

Reply:
I think I've adapted the test correctly....